README.md

```diff
@@ -220,6 +220,18 @@ It is also possible to cache gems manually, but this is not recommended because
 There are many concerns which means using `actions/cache` is never enough for caching gems (e.g., incomplete cache key, cleaning old gems when restoring from another key, correctly hashing the lockfile if not checked in, OS versions, ABI compatibility for `ruby-head`, etc).
 So, please use `bundler-cache: true` instead and report any issue.
 
+### ASan Debugging
+
+Ruby can be built with AddressSanitizer (ASan) for debugging memory issues. You can use the `asan` variant of Ruby for this:
+
+```yaml
+    - uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: asan
+```
+
+You should also compile your own code with ASan to get the full benefits of this feature.
```

Comment:
It will be automatic for extensions yes. If you’re linking in some other library maybe you want to compile that with ASAN too but that’s probably niche enough to to mention here?